### PR TITLE
feat: support DDL show cluster to display cluster info

### LINKED ISSUE #452: support DDL like show cluster to display cluster info
### Is your feature request related to a problem? Please describe.（如果该特性或者功能和已知某个问题相关，请描述问题是什么）

```bash
> show cluster

name: meta
+---------------------+--------+----------------+--------+----------+
| time                | status |   hostname     | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1702430275021313000 |   1    | 127.0.0.1:8400 |    5   |   Data   |
+---------------------+--------+----------------+--------+----------+
5 columns, 1 rows in set
```

> the `time` is `now()`
> the `nodeType` is optional for `Data` or `Meta`
> Got the data from the metaclient cacheData

```bash
> show cluster where nodeID = 5

name: meta
+---------------------+--------+----------------+--------+----------+
| time                | status |   hostname     | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1702430275021313000 |   1    | 127.0.0.1:8400 |    5   |   Data   |
+---------------------+--------+----------------+--------+----------+
5 columns, 1 rows in set
```

```bash
> show cluster where nodeType = Data

name: meta
+---------------------+--------+----------------+--------+----------+
| time                | status |   hostname     | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1702430275021313000 |   1    | 127.0.0.1:8400 |    5   |   Data   |
+---------------------+--------+----------------+--------+----------+
5 columns, 1 rows in set
```

### Describe the solution you'd like（问题的解决办法是什么）

Got the data from the metaclient cacheData.

https://github.com/openGemini/openGemini/blob/32af8ec4b051e2114a33d0474674e8e24aa816b9/lib/metaclient/meta_client.go#L328-L329

### Describe alternatives you've considered（描述您考虑过的替代方案）

_No response_

### Additional context（其他补充内容）

_No response_

### PR DESCRIPTION
**Pull request by xmh1011**
### What problem does this PR solve?

Issue Number: resolve #452 

### What is changed and how it works?
Please describe how it works
```
➜  openGemini git:(cluster-status) ts-cli
2024/01/04 17:18:53 proto: duplicate proto type registered: influxql.Measurements
2024/01/04 17:18:53 proto: duplicate proto type registered: influxql.Measurement
openGemini CLI 0.1.0 (rev-revision)
Please use `quit`, `exit` or `Ctrl-D` to exit this program.
> show cluster
+---------------------+--------+----------------+--------+----------+
| time                | status | hostname       | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1704359936579799000 | 0      | 127.0.0.3:8091 | 1      | meta     |
| 1704359936579799000 | 0      | 127.0.0.1:8091 | 2      | meta     |
| 1704359936579799000 | 0      | 127.0.0.2:8091 | 3      | meta     |
| 1704359936579799000 | 1      | 127.0.0.1:8400 | 4      | data     |
| 1704359936579799000 | 1      | 127.0.0.2:8400 | 5      | data     |
| 1704359936579799000 | 1      | 127.0.0.3:8400 | 6      | data     |
+---------------------+--------+----------------+--------+----------+
5 columns, 6 rows in set

> show cluster where nodeID = 1
+---------------------+--------+----------------+--------+----------+
| time                | status | hostname       | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1704359951958993000 | 0      | 127.0.0.3:8091 | 1      | meta     |
+---------------------+--------+----------------+--------+----------+
5 columns, 1 rows in set

> show cluster where nodeID = 1 and nodeType = meta
+---------------------+--------+----------------+--------+----------+
| time                | status | hostname       | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1704359960034243000 | 0      | 127.0.0.3:8091 | 1      | meta     |
+---------------------+--------+----------------+--------+----------+
5 columns, 1 rows in set

> show cluster where nodeType = meta
+---------------------+--------+----------------+--------+----------+
| time                | status | hostname       | nodeID | nodeType |
+---------------------+--------+----------------+--------+----------+
| 1704359965726427000 | 0      | 127.0.0.3:8091 | 1      | meta     |
| 1704359965726427000 | 0      | 127.0.0.1:8091 | 2      | meta     |
| 1704359965726427000 | 0      | 127.0.0.2:8091 | 3      | meta     |
+---------------------+--------+----------------+--------+----------+
5 columns, 3 rows in set

> show cluster where nodeType = mda
ERR: invalid node type%!(EXTRA string=mda)
> show cluster where nodeID = de
ERR: error parsing query: Invalid where clause
```

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [ ] Test B
- [ ] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open_src/influx/meta/data.go [hunk 1]
```diff
@@ -1370,6 +1384,45 @@ func (data *Data) ShowRetentionPolicies(database string) (models.Rows, error) {
 	return []*models.Row{row}, nil
 }
 
+func (data *Data) ShowClusterStatus() models.Rows {
+	row := &models.Row{Columns: []string{"time", "status", "hostname", "nodeID", "nodeType"}}
+	timestamp := time.Now().Unix()
```
Comment (shilinlee) [inline open_src/influx/meta/data.go:1389]:
```suggestion
	timestamp := time.Now().UTC().Unix()
```

File: open_src/influx/meta/data.go [hunk 2]
```diff
@@ -1370,6 +1384,45 @@ func (data *Data) ShowRetentionPolicies(database string) (models.Rows, error) {
...
+		row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, METANODE})
+	})
+	data.WalkDataNodes(func(node *DataNode) {
+		row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, DATANODE})
+	})
+
+	return []*models.Row{row}
+}
+
+func (data *Data) ShowClusterStatusWithCondition(nodeType, nodeID string, ID uint64) (models.Rows, error) {
```
Comment (shilinlee) [inline open_src/influx/meta/data.go:1401]: `nodeID` and `ID` is the same thing?

```suggestion
func (data *Data) ShowClusterStatusWithCondition(nodeType string, ID uint64) (models.Rows, error) {
```

File: open_src/influx/meta/data.go [hunk 3]
```diff
@@ -1370,6 +1384,45 @@ func (data *Data) ShowRetentionPolicies(database string) (models.Rows, error) {
...
+	row := &models.Row{Columns: []string{"time", "status", "hostname", "nodeID", "nodeType"}}
+	timestamp := time.Now().Unix()
```
Comment (shilinlee) [inline open_src/influx/meta/data.go:1403]:
```suggestion
	timestamp := time.Now().UTC().Unix()
```

File: open_src/influx/meta/data.go [hunk 4]
```diff
@@ -1370,6 +1384,45 @@ func (data *Data) ShowRetentionPolicies(database string) (models.Rows, error) {
...
+			}
+		})
+	}
+	if nodeType == DATANODE || nodeType == "" {
+		data.WalkDataNodes(func(node *DataNode) {
+			if nodeID == "" || node.ID == ID {
+				row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, DATANODE})
+			}
+		})
+	}
```
Comment (shilinlee) [inline open_src/influx/meta/data.go:1418]:
```suggestion

switch nodeType:
case METANODE:
  data.WalkMetaNodes(func(node *NodeInfo) {
	if ID == 0 || node.ID == ID {
				  row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, METANODE})
case DATANODE
  data.WalkDataNodes(func(node *DataNode) {
	if ID == 0 || node.ID == ID {
				  row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, DATANODE})
			  }
default:
	data.WalkMetaNodes(func(node *NodeInfo) {
	if ID == 0 ||  node.ID == ID {
				  row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, METANODE})		  
	data.WalkDataNodes(func(node *DataNode) {
	if ID == 0 || node.ID == ID {
				  row.Values = append(row.Values, []interface{}{timestamp, node.Status, node.Host, node.ID, DATANODE})
			  }
	
```

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/influxql/ast.go [hunk 5]
```diff
@@ -7393,6 +7393,29 @@ func (s *SetConfigStatement) String() string {
 	return buf.String()
 }
 
+type ShowClusterStatusStatement struct {
+	Condition Expr
+}
```
Comment (shilinlee) [inline open_src/influx/influxql/ast.go:7398]:
```suggestion
type ShowClusterStatusStatement struct {
	NodeType string
	NodeID   string
}
```

File: open_src/influx/influxql/sql.y [hunk 6]
```diff
@@ -3284,4 +3288,12 @@ SET_CONFIG_STATEMENT:
...
     }
 
+SHOW_CLUSTER_STATUS_STATEMENT:
+    SHOW CLUSTER STATUS WHERE_CLAUSE
+     {
+          stmt := &ShowClusterStatusStatement{}
+          stmt.Condition = $4
+          $$ = stmt
+     }
+
```
Comment (shilinlee) [inline open_src/influx/influxql/sql.y:3298]:
```suggestion
SHOW_CLUSTER_STATUS_STATEMENT:
    SHOW CLUSTER STATUS WHERE_CLUSTER_CLAUSE
     {
          $$ = $4
     }

WHERE_CLUSTER_CLAUSE:
    WHERE IDENT EQ IDENT
    {
        stmt := &ShowClusterStatusStatement{}
        if strings.ToLower($2) == "nodetype" {
            stmt.NodeType = $4
        } else {
            stmt.NodeID = $4
        }
        $$ = stmt
    }
    |WHERE IDENT EQ IDENT AND IDENT EQ IDENT
    {
        stmt := &ShowClusterStatusStatement{}
        if strings.ToLower($2) == "nodetype" {
            stmt.NodeType = $4
        } else {


            stmt.NodeID = $4
        }
        if strings.ToLower($6) == "nodetype" {
            stmt.NodeType = $8
        } else {
            stmt.NodeID = $8
        }
        $$ = stmt
    }
    |
    {
        $$ = &ShowClusterStatusStatement{}
    }

```

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/influxql/sql.y [hunk 7]
```diff
@@ -3284,4 +3288,12 @@ SET_CONFIG_STATEMENT:
         $$ = stmt
     }
 
+SHOW_CLUSTER_STATUS_STATEMENT:
+    SHOW CLUSTER STATUS WHERE_CLAUSE
```
Comment (shilinlee) [inline open_src/influx/influxql/sql.y:3292]:
```suggestion
    SHOW CLUSTER WHERE_CLAUSE
```

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — APPROVED** [review]

**Review by fx408 — APPROVED** [review]
ok

**Review by scuzyj — APPROVED** [review]

**Comment by scuzyj** [comment]
image
it is better to convert status to a string，because the meanings of numbers cannot be understood.

**Pull request merged**